lib/std/crypto/aes/soft.zig

```diff
@@ -0,0 +1,761 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+// Based on Go stdlib implementation
```

Comment:
This file is just begging to be rewritten with `@Vector` and zig's architecture-independent SIMD. That should probably be a separate PR though :)

Reply:
Yes, even though, really, it should be replaced with a constant-time implementation rather than use lookup tables.

Reply:
Speed gains are possible just with permutations (see Mike Hamburg's [Accelerating AES with Vector Permute Instructions](https://www.shiftleft.org/papers/vector_aes/vector_aes.pdf) paper), that Zig supports (`@shuffle`).

Unfortunately, permutations are not very efficient on ARM, and emulation on CPUs without dedicated instructions is painfully slow, especially if they have to be constant-time.

So, this technique would really be only useful on Intel CPUs without AES-NI, and only for short messages, since even on these CPUs, bitsliced/fixsliced implementations are faster.